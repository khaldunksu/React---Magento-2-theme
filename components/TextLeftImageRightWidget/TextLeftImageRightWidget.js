import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ReactHtmlParser from "react-html-parser";
import "./TextLeftImageRightWidget.scss";

export function TextLeftImageRightWidget(props) {
    const { data } = props;

    return (
        <section className="text-left-widget js-hook__text-left-widget">
            <img className="js-hook__text-left-widget-image" src={data.image} alt=""/>
            <div className="js-hook__text-left-widget-description">{ReactHtmlParser(data.text)}</div>
        </section>
    );
}

TextLeftImageRightWidget.propTypes = {};
TextLeftImageRightWidget.defaultProps = {};

export default connect(state => state)(TextLeftImageRightWidget);
