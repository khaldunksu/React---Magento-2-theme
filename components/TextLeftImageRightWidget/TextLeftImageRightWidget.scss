.text-left-widget{
  background: url("/static/img/main-home-bg.png");
  background-repeat: no-repeat;
  background-size: contain;
  background-repeat: no-repeat;
  background-size: contain;
  flex-direction: column;
  justify-content: center;
  align-items: flex-end;

  &__content{
    width: 50%;
    max-height: 50%;
  }

  &__image{
    display: block;
    max-width: 100%;
    max-height: 100%;
  }
}