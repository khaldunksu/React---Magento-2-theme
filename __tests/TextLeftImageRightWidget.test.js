import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import { Navigation } from "../components/TextLeftImageRightWidget/TextLeftImageRightWidget";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;

describe("Component Navigation matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<TextLeftImageRightWidget />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component TextLeftImageRightWidget", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<TextLeftImageRightWidget />);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
        expect(wrapper.find(".js-hook__text-left-widget").length).toBe(1);
    });

});