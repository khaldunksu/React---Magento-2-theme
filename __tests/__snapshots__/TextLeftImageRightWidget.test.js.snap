// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Navigation matches snapshot renders page layout correctly 1`] = `
<HomeSection
  classNames="text-left-widget js-hook__text-left-widget"
  contentClass="text-left-widget__content"
>
  <img
    alt=""
    className="js-hook__text-left-widget-image text-left-widget__image"
    src="some-test-url"
  />
  <div
    className="js-hook__text-left-widget-description text-left-widget__description"
  >
    Test text
  </div>
</HomeSection>
`;
